### PR TITLE
Add flag to disable request stat recording

### PR DESCRIPTION
**Pull request by cberner**

### CODE REVIEW

File: http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java [hunk 1]
```diff
@@ -46,6 +46,19 @@
     private HostAndPort socksProxy;
     private String keyStorePath = System.getProperty(JAVAX_NET_SSL_KEY_STORE);
     private String keyStorePassword = System.getProperty(JAVAX_NET_SSL_KEY_STORE_PASSWORD);
+    private boolean recordRequestStats = true;
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java]: Move the field to the top, so the field and getter/setter order matches.

**Comment by electrum** [comment]
This looks fine, but I don't like the idea of completely disabling stats. I'm thinking we could have a "simple" request stats object that keeps at least a request count. It could track total time and bytes, though those would eventually overflow.

**Comment by dain** [comment]
I think @martint should figure out why the stats are so expensive.

**Comment by cberner** [comment]
I filed https://github.com/airlift/airlift/issues/321 . I'll hold off on merging this

**Comment by dain** [comment]
Superseded by #323

**Pull request closed without merging**